swarmrl/networks/flax_network.py

```diff
@@ -217,22 +219,23 @@ def restore_model_state(self, filename, directory):
         )
         self.epoch_count = epoch
 
-    def __call__(self, feature_vector: np.ndarray):
+    def __call__(self, parmas: FrozenDict, episode_features):
         """
-        See parent class for full doc string.
+        vmape
 
         Parameters
         ----------
-        feature_vector : np.ndarray
-                Observable to be passed through the network on which a decision is made.
+        parmas : dict
+                Parameters of the model.
+        episode_features:
+                Features of the episode. This contains the features of all agents,
+                for all time steps in the episode.
+
 
         Returns
         -------
         logits : np.ndarray
                 Output of the network.
         """
 
-        try:
-            return self.apply_fn({"params": self.model_state.params}, feature_vector)
-        except AttributeError:  # We need this for loaded models.
-            return self.apply_fn({"params": self.model_state["params"]}, feature_vector)
+        return self.batch_apply_fn({"params": parmas}, episode_features)
```

Comment:
Why don't you use the apply function here? It would be faster.

Reply:
Because it computes the logits for the whole episode. 
With that, we need to call the apply fn in the loss only once.

Reply:
Okay.